ext/hook/uhook_attributes.stub.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+/** @generate-class-entries */
+
+namespace DDTrace;
+
+require "Zend/zend_attributes.stub.php";
+
+#[\Attribute(\Attribute::TARGET_FUNCTION | \Attribute::TARGET_METHOD)]
+final class Traced {
+    public function __construct(
+        string $name = "",
+        string $resource = "",
+        string $type = "",
+        string $service = "",
+        array $tags = [],
+        bool|array $args = false,
+        bool $return = false,
+        bool $recurse = true,
+        bool $run_if_limited = false
+    ) {}
+}
```

Comment:
I think I understand most of these options, but not `$args` nor `$return`. What are these?

Also, won't the linter complain about `$run_if_limited`? IIRC it likes `$runIfLimited`.

Reply:
And just to be sure I'd add to the list `$recurse`, `$run_if_limited`

Reply:
$args and $return log the args and return value into tags. I.e. `$args = ["foo"]` will add a tag `arg.foo = <a possibly shortened representation of the passed parameter>`. (`$args = true` will log them all).

Reply:
@morrisonlevi The linter does not complain in fact :-D It's named that way because the array option for trace_function() is also named that way. Won't object to changing, thought it'd be more consistent.

Reply:
Thanks. And the api looks good to me now that you've renamed a few things.